Mathlib/Tactic/Rewrites.lean

```diff
@@ -118,15 +118,19 @@ structure RewriteResult where
   /-- Can the new goal in `result` be closed by `with_reducible rfl`? -/
   -- This is an `Option` so that it can be computed lazily.
   rfl? : Option Bool
+  /-- The metavariable context after the rewrite.
+  This needs to be stored as part of the result so we can backtrack the state. -/
+  mctx : MetavarContext
 
 /-- Update a `RewriteResult` by filling in the `rfl?` field if it is currently `none`,
 to reflect whether the remaining goal can be closed by `with_reducible rfl`. -/
 def RewriteResult.computeRfl (r : RewriteResult) : MetaM RewriteResult := do
```

Comment:
```suggestion
def RewriteResult.computeRfl (r : RewriteResult) : MetaM RewriteResult := withReducible do
```
I think we need a `withReducible` in front of this and probably in front of `evalTactic (← `(tactic| try rfl))` on line 267 (or simply in front of the whole tactic, but I'm not sure if that would change the behavior somehow). A test:
```
def zero : Nat := 0

example : zero = 0 := by rw? -- Try this: rw [← Nat.zero_eq] -- "no goals"

-- After applying the suggestion:
example : zero = 0 := by rw [← Nat.zero_eq] -- unsolved goals ⊢ zero ≤ 0

-- Contrast with:
example : 0 = 0 := by rw [← Nat.zero_eq] -- (goal closed)
```
Again, maybe for a different PR, but noticed it while thinking about `computeRfl`. :)

Reply:
Thanks. Fixed, and added test.